### PR TITLE
Add parallel transport for the affine invariant metric on SPDs

### PR DESCRIPTION
**Pull request by nguigs**
Also add a congruence operator in matrices for convenience.
Doesn't work with tf yet because of the sqrtm method, but this can be fixed with tf 2.

### CODE REVIEW

**Comment by nguigs** [comment]
Tests failed for python 3.5 to 3.7 with pytorch, but passed for pyton 3.8

File: geomstats/geometry/matrices.py [hunk 1]
```diff
@@ -132,6 +132,23 @@ def random_uniform(self, n_samples=1):
         point = gs.random.rand(n_samples, self.m, self.n)
         return point
 
+    @classmethod
+    def cong(cls, mat_1, mat_2):
+        """Compute the congruent action of mat_2 on mat_1.
+
+        This is `mat_2 mat_1 mat_2^T`
```
Comment (ninamiolane) [inline geomstats/geometry/matrices.py]: Use :math: syntax. Also, a description needs to end with a . as per : https://numpydoc.readthedocs.io/en/latest/format.html

File: geomstats/geometry/matrices.py [hunk 2]
```diff
@@ -132,6 +132,23 @@ def random_uniform(self, n_samples=1):
...
+
+        Parameters
+        ----------
+        mat_1 : array-like, shape=[n_samples, n, n]
```
Comment (ninamiolane) [inline geomstats/geometry/matrices.py]: Missing parameters' descriptions. See conversation on slack for documentation's conventions.

File: geomstats/geometry/spd_matrices.py [hunk 3]
```diff
@@ -623,6 +623,35 @@ def geodesic(self, initial_point, initial_tangent_vec):
...
+
+        Closed-form solution for the parallel transport of a tangent vector a
+        along the geodesic defined by exp_(base_point)(tangent_vec_b).
+        Denoting `tangent_vec_a` by `S`, `base_point` by `A`, let
+        `B = Exp_A(tangent_vec_b)` and :math: `E = (BA^{- 1})^({ 1 / 2})`.
+        Then the
+        parallel transport to `B`is:
+
+        ..math::
+                        S' = ESE^T
```
Comment (ninamiolane) [inline geomstats/geometry/spd_matrices.py]: Missing . at the end of description

File: geomstats/geometry/spd_matrices.py [hunk 4]
```diff
@@ -623,6 +623,35 @@ def geodesic(self, initial_point, initial_tangent_vec):
             initial_tangent_vec=initial_tangent_vec,
             point_type='matrix')
 
+    def parallel_transport(self, tangent_vec_a, tangent_vec_b, base_point):
```
Comment (ninamiolane) [inline geomstats/geometry/spd_matrices.py]: 🎉 !

File: geomstats/geometry/spd_matrices.py [hunk 5]
```diff
@@ -623,6 +623,35 @@ def geodesic(self, initial_point, initial_tangent_vec):
...
+        `B = Exp_A(tangent_vec_b)` and :math: `E = (BA^{- 1})^({ 1 / 2})`.
+        Then the
+        parallel transport to `B`is:
+
+        ..math::
+                        S' = ESE^T
+
+        Parameters
+        ----------
+        tangent_vec_a : array-like, shape=[n_samples, dimension + 1]
```
Comment (ninamiolane) [inline geomstats/geometry/spd_matrices.py]: Missing parameters' description. I have filled one on parallel_transport in PR #425 , maybe copy paste this one?

File: geomstats/geometry/matrices.py [hunk 6]
```diff
@@ -132,6 +132,23 @@ def random_uniform(self, n_samples=1):
         point = gs.random.rand(n_samples, self.m, self.n)
         return point
 
+    @classmethod
+    def cong(cls, mat_1, mat_2):
```
Comment (ninamiolane) [inline geomstats/geometry/matrices.py]: call it congruent? I know it's longer, but it's more explicit?

File: tests/test_matrices.py [hunk 7]
```diff
@@ -167,3 +167,21 @@ def test_inner_product(self):
         expected = helper.to_scalar(expected)
 
         self.assertAllClose(result, expected)
+
+    def test_cong(self):
```
Comment (ninamiolane) [inline tests/test_matrices.py]: test_congruent

**Comment by nguigs** [comment]
#448 created

**Pull request merged**